src/viam/app/data_client.py

```diff
@@ -158,6 +158,12 @@ class TabularDataPoint:
         resource_name: str
         """The resource name"""
 
+        resource_subtype: str
+        """
+        deprecated; use `resource_api`
+        The resource API. Ex: `rdk:component:sensor`
+        """
```

Comment:
(minor) I'd love for this to be a property instead, so people can't set it, it's impossible for it to ever drift from the value in `resource_api`, and so we can emit a deprecation warning on call instead of just including it in comments. Something like

```python
@property
def resource_subtype(self) -> str:
    warnings.warn("<some deprecation warning>", DeprecationWarning, stacklevel=2)
    return self.resource_api
```